lib/models/mirror.js

```diff
@@ -38,13 +43,27 @@ Mirror.set('toObject', {
 
 Mirror.statics.create = function(contract, contact, callback) {
   var self = this;
+
+  if (_.isEmpty(contract)) {
+    return callback(new Error('Cannot save mirror: invalid contract'));
+  }
+
+  if (_.isEmpty(contact)) {
+    return callback(new Error('Cannot save mirror: invalid contact'));
+  }
+
   var mirror = new self({
     shardHash: contract.data_hash,
     contact: contact.nodeID,
     contract: contract
   });
 
-  mirror.save(callback);
+  mirror.save(function(err, mirror) {
+    if (err) {
+      return callback(new Error('Cannot save mirror', err));
```

Comment:
Mongoose's .save method returns the err and then the newly saved doc as the second argument. I'm not really sure why it's not returned in the callback.